enhancements/dns/dns-operator-operand-logging-level.md

```diff
@@ -0,0 +1,257 @@
+
+Valid values for coredns logLevel are: "Normal", "Debug", "Trace" as per `DNSLogLevel` type.
+
+We will enable logging of CoreDNS's [classes of responses](https://github.com/coredns/coredns/tree/master/plugin/log#syntax) that correspond to the log level specified in the API.
+So,
+
+`logLevel: "Normal"`  will enable the "errors" class: `log . { class error }`.
+
+`logLevel: "Debug"` will enable the "denial" class: `log . { class denial error }`.
+
+`logLevel: "Trace"` will enable the "all" class: `log . { class all }`.
+
+Note that the `errors` plugin is always enabled.  The `errors` plugin logs TCP/UDP connection errors whereas `log . { class error }` logs DNS error responses (such as SERVFAIL).
+The CoreDNS reloads its configuration without requiring a restart, so the operator can adjust CoreDNS's log level just by updating the Corefile configmap without need to restart the pod.
+
+
+We will be adding an API field `operatorlogLevel` in `DNSSpec` with the type `DNSLogLevel`:
+```go
+// operatorLogLevel controls the logging level of the DNS Operator.
+// Valid values are: "Normal", "Debug", "Trace".
+// Defaults to "Normal".
+// setting operatorLogLevel: Trace will produce extremely verbose logs.
+// +optional
+// +kubebuilder:default=Normal
+OperatorLogLevel DNSLogLevel `json:"operatorLogLevel,omitempty"`
+```
+This new field would allow a cluster administrator to specify the desired logging level specifically for the DNS Operator.
+
+Additionally, a new API field `LogLevel` of type `DNSLogLevel` will be added to specify the log level for CoreDNS:
+```go
+// logLevel describes the desired logging verbosity for CoreDNS.
+// Any one of the following values may be specified:
+// * Normal logs errors from upstream resolvers.
+// * Debug logs errors, NXDOMAIN responses, and NODATA responses.
+// * Trace logs errors and all responses.
+//  Setting logLevel: Trace will produce extremely verbose logs.
+// Valid values are: "Normal", "Debug", "Trace".
+// Defaults to "Normal".
+// +optional
+// +kubebuilder:default=Normal
+LogLevel DNSLogLevel `json:"logLevel,omitempty"`
+```
+
+Both of these new API fields use the aforementioned `DNSLogLevel` type, which is defined as follows:
+```go
+
+// +kubebuilder:validation:Enum:=Normal;Debug;Trace
+type DNSLogLevel string
+
+var (
+// Normal is the default.  Normal, working log information, everything is fine, but helpful notices for auditing or common operations.  In kube, this is probably glog=2.
+DNSLogLevelNormal DNSLogLevel = "Normal"
+
+// Debug is used when something went wrong.  Even common operations may be logged, and less helpful but more quantity of notices.  In kube, this is probably glog=4.
+DNSLogLevelDebug DNSLogLevel = "Debug"
+
+// Trace is used when something went really badly and even more verbose logs are needed.  Logging every function call as part of a common operation, to tracing execution of a query.  In kube, this is probably glog=6.
+DNSLogLevelTrace DNSLogLevel = "Trace"
+)
+
+```
```

Comment:
The definitions we ended up with in https://github.com/openshift/api/pull/1031 look a bit different; the enhancement text should be updated eventually.